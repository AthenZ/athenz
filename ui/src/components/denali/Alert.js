/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import Icon from '../denali/icons/Icon';
import {
    colors,
    cssDropShadow,
    cssFontStyles,
    cssFontWeights,
} from '../denali/styles';
import { css, cx } from 'emotion';
import { rgba } from 'polished';

const TYPE_COLOR_MAP = {
    danger: 'statusDanger',
    info: 'statusInfo',
    success: 'statusSuccess',
    warning: 'statusWarning',
};

const TYPE_ICON_MAP = {
    danger: 'stop-warning',
    info: 'information-circle',
    success: 'check-circle',
    warning: 'warning',
};

// http://reactcommunity.org/react-modal/examples/set_app_element.html
ReactModal.setAppElement('body');

const modalOverlayClass = css`
    background-color: transparent;
    bottom: 0;
    left: 0;
    position: fixed;
    right: 0;
    top: 0;
    z-index: 5;
    label: modal-overlay;
`;

const makeAlertClass = (props) => css`
    ${cssDropShadow};
    align-items: flex-start;
    background: ${colors.white};
    border-radius: 4px;
    border-top: 4px solid ${colors[TYPE_COLOR_MAP[props.type]]};
    box-sizing: border-box;
    color: ${colors.black};
    display: flex;
    flex-flow: row nowrap;
    ${cssFontStyles.default};
    left: 50%;
    max-height: calc(100vh - 160px);
    min-width: 400px;
    outline: none;
    position: fixed;
    transform: translate(-50%, 0);
    width: 400px;
    label: alert;
    &.ReactModal__Content {
        opacity: 0;
        top: 0;
    }
    &.animated {
        transition: opacity 0.2s ease-out,
            top 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275);
    }
    &.ReactModal__Content--after-open {
        top: 80px;
        opacity: 1;
    }
    /* Animate fade out on close. This is currently not used as tests fail */
    &.ReactModal__Content--before-close {
        opacity: 0;
    }
    & .left-section {
        flex: 0 0 28px;
        padding: 14px;
        & .alert-icon {
            align-items: flex-start;
            display: flex;
            justify-content: center;
        }
    }
    & .right-section {
        display: flex;
        flex: 1 1 auto;
        flex-flow: column nowrap;
        & .title-row {
            align-items: center;
            display: flex;
            flex-flow: row nowrap;
            height: initial;
            justify-content: space-between;
            padding: 14px 14px 14px 0;
            & .title {
                ${cssFontWeights.bold};
            }
        }
        & .description {
            font-size: 14px;
            font-weight: 300;
            padding: 0 14px 14px 0;
        }
    }
`;

/**
 * Alert displays a dismissable modal near the top of the screen. It is meant to
 * show a short blurb of information. There are four
 * alert types:
 *
 * 1. Info
 * 2. Success
 * 3. Warning
 * 4. Danger
 *
 * Under the hood, Alert uses [`react-modal`](http://reactcommunity.org/react-modal/).
 * To prevent the page from scrolling when the modal is showing, add the
 * following to the app's global CSS:
 *
 * ```css
 * .ReactModal__Body--open {
 *   overflow: hidden;
 * }
 */
class Alert extends React.Component {
    // Taken from:
    // https://stackoverflow.com/questions/45678517/better-way-to-cleartimeout-in-componentwillunmount
    setTimer = (ms) => {
        // Remember the timer handle
        this.timerHandle = setTimeout(() => {
            this.timerHandle = null;
            this.props.onClose();
        }, ms);
    };

    clearTimer = () => {
        // Is our timer running?
        if (this.timerHandle) {
            clearTimeout(this.timerHandle);
            this.timerHandle = null;
        }
    };

    componentDidMount() {
        if (this.props.isOpen && this.props.duration) {
            this.setTimer(this.props.duration);
        }
    }

    componentDidUpdate(prevProps) {
        if (!prevProps.isOpen && this.props.isOpen && this.props.duration) {
            this.setTimer(this.props.duration);
        } else if (prevProps.isOpen && !this.props.isOpen) {
            this.clearTimer();
        }
    }

    componentWillUnmount() {
        this.clearTimer();
    }

    render() {
        const StatusIcon = TYPE_ICON_MAP[this.props.type];
        const alertClass = makeAlertClass(this.props);

        return (
            <ReactModal
                aria={{
                    labelledby: 'alert-title',
                    describedby: 'alert-description',
                }}
                className={cx(
                    alertClass,
                    'denali-alert',
                    `denali-alert-${this.props.type}`,
                    {
                        animated: !this.props.noanim,
                    },
                    this.props.className
                )}
                closeTimeoutMS={200}
                descriptionLabel={`${this.props.type} alert`}
                isOpen={this.props.isOpen}
                overlayClassName={modalOverlayClass}
                shouldCloseOnOverlayClick={false}
                onRequestClose={this.props.onClose}
            >
                <div className='left-section'>
                    <div className='alert-icon' data-testid='alert-icon'>
                        <Icon
                            icon={'check-circle'}
                            color={colors[TYPE_COLOR_MAP[this.props.type]]}
                            decorative={true}
                            size='28px'
                            verticalAlign='middle'
                        />
                    </div>
                </div>
                <div className='right-section'>
                    <div className='title-row'>
                        <div
                            className='title'
                            id='alert-title'
                            data-testid='alert-title'
                        >
                            {this.props.title}
                        </div>
                        <div
                            className='close-button'
                            onClick={this.props.onClose}
                        >
                            <Icon
                                color={rgba(colors.grey800, 0.4)}
                                colorHover={rgba(colors.grey800, 0.6)}
                                isLink
                                size='28px'
                                icon={'close'}
                            />
                        </div>
                    </div>
                    {this.props.description && (
                        <div
                            className='description'
                            id='alert-description'
                            data-testid='alert-description'
                        >
                            {this.props.description}
                        </div>
                    )}
                </div>
            </ReactModal>
        );
    }
}

Alert.propTypes = {
    /** Additonal class to apply to the outer div of the modal content */
    className: PropTypes.string,
    /** Optional description. Can be a string or JSX / component */
    description: PropTypes.any,
    /** Once open, call `onClose()` after N milliseconds */
    duration: PropTypes.number,
    /** Disable animations / transitions */
    noanim: PropTypes.bool,
    /** Toggle the alert on or off */
    isOpen: PropTypes.bool,
    /** Title of the alert */
    title: PropTypes.string.isRequired,
    /** Alert supports four status types */
    type: PropTypes.oneOf(['danger', 'info', 'success', 'warning']),
    /** Handler for closing the alert */
    onClose: PropTypes.func.isRequired,
};

Alert.defaultProps = {
    duration: 0,
    isOpen: false,
    noanim: false,
    type: 'info',
};

export default Alert;
