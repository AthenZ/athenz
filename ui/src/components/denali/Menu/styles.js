/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { css } from 'emotion';
import { cssFontStyles } from '../styles/fonts';
import { colors } from '../styles/colors';
import { cssDropShadow } from '../styles/drop-shadow';

export const makeCssPopperBox = (props) => css`
    ${cssDropShadow};
    ${cssFontStyles.default};
    background-color: ${colors.white};
    border: 2px solid ${colors.border};
    border-radius: 4px;
    color: ${colors.black};
    cursor: default;
    line-height: 1.4;
    margin: 0;
    max-width: ${props.basic ? '400px' : undefined};
    min-width: 20px;
    padding: ${props.basic ? '6px 10px' : 0};
    transition: ${!props.noanim && 'opacity 0.25s ease-in'};
    z-index: 999;
    label: denali-menu;
    /* Shift the popper to allow space for arrow */
    &[data-placement*='bottom'] {
        margin-top: 8px;
    }
    &[data-placement*='top'] {
        margin-bottom: 8px;
    }
    &[data-placement*='left'] {
        margin-right: 8px;
    }
    &[data-placement*='right'] {
        margin-left: 8px;
    }

    /**
   * Invisible padding between trigger and main content (around the arrow) so
   * user can slide from one to the other without the menu closing.
   */
    &::after {
        background: transparent;
        content: '';
        position: absolute;
    }
    &[data-placement*='top']::after {
        bottom: -12px;
        height: 12px;
        width: 100%;
    }
    &[data-placement*='bottom']::after {
        height: 12px;
        top: -12px;
        width: 100%;
    }
    &[data-placement*='left']::after {
        right: -12px;
        height: 100%;
        width: 12px;
    }
    &[data-placement*='right']::after {
        left: -12px;
        height: 100%;
        width: 12px;
    }
`;

export const cssArrow = css`
    height: 1rem;
    position: absolute;
    width: 1rem;
    label: menu-arrow;

    &::before,
    &::after {
        border-style: solid;
        content: '';
        display: block;
        height: 0;
        margin: auto;
        position: absolute;
        width: 0;
    }

    &[data-placement*='bottom'] {
        left: 0;
        margin-top: -6px;
        top: -8px;
        width: 1rem;
    }

    &[data-placement*='bottom']::before {
        border-color: transparent transparent ${colors.grey400} transparent;
        border-width: 0 8px 8px 8px;
        position: absolute;
        top: 4px;
    }

    &[data-placement*='bottom']::after {
        border-color: transparent transparent ${colors.white} transparent;
        border-width: 0 8px 8px 8px;
        top: 7px;
    }

    &[data-placement*='top'] {
        bottom: 0;
        height: 1rem;
        left: 0;
        margin-bottom: -1rem;
        width: 1rem;
    }

    &[data-placement*='top']::before {
        border-color: ${colors.grey400} transparent transparent transparent;
        border-width: 8px 8px 0 8px;
        position: absolute;
        top: 2px;
    }

    &[data-placement*='top']::after {
        border-color: ${colors.white} transparent transparent transparent;
        border-width: 8px 8px 0 8px;
        top: -1px;
    }

    &[data-placement*='right'] {
        height: 1rem;
        left: 0;
        margin-left: -1rem;
        width: 1rem;
    }

    &[data-placement*='right']::before {
        border-color: transparent ${colors.grey400} transparent transparent;
        border-width: 8px 8px 8px 0;
        left: 5px;
    }

    &[data-placement*='right']::after {
        border-color: transparent ${colors.white} transparent transparent;
        border-width: 8px 8px 8px 0;
        left: 8px;
        top: 0;
    }

    &[data-placement*='left'] {
        height: 1rem;
        margin-right: -1rem;
        right: 0;
        width: 1rem;
    }

    &[data-placement*='left']::before {
        border-color: transparent transparent transparent ${colors.grey400};
        border-width: 8px 0 8px 8px;
        left: 2px;
    }

    &[data-placement*='left']::after {
        border-color: transparent transparent transparent ${colors.white};
        border-width: 8px 0 8px 8px;
        left: -1px;
        top: 0;
    }
`;
