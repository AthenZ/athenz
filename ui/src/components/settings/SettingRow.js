/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import styled from '@emotion/styled';
import Switch from '../denali/Switch';
import Input from '../denali/Input';
import InputDropdown from '../denali/InputDropdown';

const TDStyled = styled.td`
    background-color: ${(props) => props.color};
    text-align: ${(props) => props.align};
    padding: 5px 0 5px 15px;
    vertical-align: middle;
    word-break: break-all;
`;

const TrStyled = styled.tr`
    box-sizing: border-box;
    margin-top: 10px;
    box-shadow: 0 1px 4px #d9d9d9;
    border: 1px solid #fff;
    -webkit-border-image: none;
    border-image: none;
    -webkit-border-image: initial;
    border-image: initial;
    height: 50px;
`;

const StyledDiv = styled.div`
    display: flex;
`;

const SettingInput = styled(Input)`
    margin-top: 5px;
`;

const StyledInputDropDown = styled(InputDropdown)`
    margin-top: 5px;
    display: block;
`;

export default class SettingRow extends React.Component {
    constructor(props) {
        super(props);
        this.onTimeChange = this.onTimeChange.bind(this);
        this.onDropDownChange = this.onDropDownChange.bind(this);
        this.toggleSwitchButton = this.toggleSwitchButton.bind(this);
        this.onRadioChange = this.onRadioChange.bind(this);
        this.saveJustification = this.saveJustification.bind(this);
        this.api = props.api;
    }

    saveJustification(val) {
        this.setState({ deleteJustification: val });
    }

    toggleSwitchButton(evt) {
        this.props.onValueChange(this.props.name, evt.currentTarget.checked);
    }

    onTimeChange(evt) {
        this.props.onValueChange(this.props.name, evt.target.value);
    }

    onDropDownChange(evt) {
        let value = evt ? evt.value : '';
        this.props.onValueChange(this.props.name, value);
    }

    onRadioChange(event) {
        if (event.target.value) {
            this.props.onValueChange(this.props.name, event.target.value);
        }
    }

    numRestricted(event) {
        const re = /[0-9]+/g;
        if (!re.test(event.key)) {
            event.preventDefault();
        }
    }

    getSettingButton() {
        switch (this.props.type) {
            case 'switch':
                return (
                    <Switch
                        name={'setting' + this.props.name}
                        value={this.props.value}
                        checked={this.props.value}
                        onChange={this.toggleSwitchButton}
                    />
                );
            case 'input':
                return (
                    <StyledDiv>
                        <SettingInput
                            pattern='[0-9]*'
                            placeholder={this.props.unit}
                            fluid
                            id={'setting-' + this.props.name}
                            onChange={this.onTimeChange}
                            onKeyPress={this.numRestricted}
                            value={this.props.value}
                        />
                    </StyledDiv>
                );
            case 'dropdown':
                return (
                    <StyledDiv>
                        <StyledInputDropDown
                            fluid
                            name={'setting-' + this.props.name}
                            options={this.props.options}
                            placeholder={this.props.placeholder}
                            filterable
                            onChange={this.onDropDownChange}
                            defaultSelectedValue={this.props.value}
                        />
                    </StyledDiv>
                );
        }
    }

    render() {
        let rows = [];
        let left = 'left';
        let color = this.props.color;
        let label = this.props.label;

        if (this.props.unit) {
            label += this.props.unit ? ` (${this.props.unit})` : '';
        }
        let name = this.props.name;

        let button = this.getSettingButton();

        rows.push(
            <TrStyled key={name} data-testid='setting-row'>
                <TDStyled color={color} align={left}>
                    {label}
                </TDStyled>
                <TDStyled color={color} align={left}>
                    {button}
                </TDStyled>
                <TDStyled color={color} align={left}>
                    {this.props.desc}
                </TDStyled>
            </TrStyled>
        );

        return rows;
    }
}
