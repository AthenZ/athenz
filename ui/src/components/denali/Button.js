/*
 * Copyright 2020 Verizon Media
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { cx, css } from 'emotion';
import { rgba } from 'polished';
import { colors } from './styles/colors';
import { cssFontFamilies, cssFontWeights } from './styles/fonts';

// NOTE: Psuedo class priority:
// Link Visited Hover Focus Active
// "Lord Vader Hates Furry Animals"

const buttonBase = (props) => css`
    border: none;
    border-radius: 2px;
    box-shadow: 0 0 0 1px transparent inset, 0 0 0 0 ${colors.grey500} inset;
    box-sizing: border-box;
    cursor: pointer;
    display: inline-block;
    ${cssFontFamilies.default};
    ${cssFontWeights.normal};
    font-size: ${props.size === 'large'
        ? '18px'
        : props.size === 'regular'
        ? '14px'
        : '12px'};
    line-height: 1;
    margin: 5px;
    min-width: ${props.size === 'small' ? '70px' : undefined};
    outline: 0;
    padding: ${props.size === 'large'
        ? '12px 28px'
        : props.size === 'regular'
        ? '10px 24px'
        : '8px 18px'};
    text-align: center;
    text-shadow: none;
    text-transform: none;
    transition: ${!props.noanim ? 'all 0.2s ease-in' : undefined};
    vertical-align: baseline;
    white-space: nowrap;
    label: button;
    &:first-of-type {
        margin-left: 0;
    }
    &:disabled {
        background: ${props.dark
            ? rgba(colors.white, 0.1)
            : rgba(colors.grey800, 0.1)};
        color: ${props.dark
            ? rgba(colors.white, 0.2)
            : rgba(colors.grey800, 0.2)};
        cursor: not-allowed;
    }
`;

const buttonVariants = {
    primary: (props) => css`
        background: linear-gradient(
            to right,
            ${props.active ? colors.brand700 : colors.brand600},
            ${props.active ? colors.brand800 : colors.brand700}
        );
        color: ${colors.white};
        &:hover:not(:disabled) {
            background: linear-gradient(
                to right,
                ${colors.brand700},
                ${colors.brand800}
            );
        }
    `,
    secondary: (props) => css`
        background: ${props.dark
            ? props.active
                ? `linear-gradient(to right, ${rgba(
                      colors.brand700,
                      0.3
                  )}, ${rgba(colors.brand800, 0.3)})`
                : 'transparent'
            : props.active
            ? colors.brand200
            : 'transparent'};
        box-shadow: 0 0 0 1px ${colors.brand600} inset;
        color: ${props.dark ? colors.white : colors.brand600};
        &:disabled {
            box-shadow: ${props.dark
                ? `0 0 0 1px ${rgba(colors.white, 0.2)} inset`
                : `0 0 0 1px ${rgba(colors.grey800, 0.1)} inset`};
        }
        &:hover:not(:disabled) {
            background: ${props.dark
                ? `linear-gradient(to right, ${rgba(
                      colors.brand700,
                      0.3
                  )}, ${rgba(colors.brand800, 0.3)})`
                : colors.brand200};
        }
    `,
    danger: () => css`
        background: ${colors.red500};
        color: ${colors.white};
        &:hover:not(:disabled) {
            background: ${colors.red600};
        }
    `,
};

class Button extends React.PureComponent {
    render() {
        const {
            active,
            className,
            danger,
            dark,
            noanim,
            secondary,
            size,
            ...rest
        } = this.props;

        const cssProps = { active, dark, noanim, size };

        const makeCssButton = (props) => css`
            ${buttonBase(props)};
            ${(danger
                ? buttonVariants.danger
                : secondary
                ? buttonVariants.secondary
                : buttonVariants.primary)(props)};
        `;
        const cssButton = makeCssButton(cssProps);

        const classes = cx(cssButton, 'denali-button', className);

        return <button className={classes} {...rest} />;
    }
}

Button.propTypes = {
    /** Set active state */
    active: PropTypes.bool,
    /** Additonal class to apply to the button */
    className: PropTypes.string,
    /** Error color scheme */
    danger: PropTypes.bool,
    /** Dark color scheme (can be used in conjunction with 'secondary') */
    dark: PropTypes.bool,
    /** Disable the button (no clicks) */
    disabled: PropTypes.bool,
    /** Forward React ref to the `<input>` element */
    innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
    /** Name of the `button` element */
    name: PropTypes.string,
    /** Disable animations / transitions */
    noanim: PropTypes.bool,
    /** Secondary color scheme */
    secondary: PropTypes.bool,
    /** Size of buttons (default is regular) */
    size: PropTypes.oneOf(['large', 'regular', 'small']),
    /** <button> type attribute */
    type: PropTypes.oneOf(['button', 'submit', 'reset']),
    /** <button> value attribute */
    value: PropTypes.any,
    /** Gets called when the button is clicked */
    onClick: PropTypes.func,
};

Button.defaultProps = {
    active: false,
    danger: false,
    dark: false,
    disabled: false,
    noanim: false,
    size: 'regular',
    secondary: false,
};

export default Button;
