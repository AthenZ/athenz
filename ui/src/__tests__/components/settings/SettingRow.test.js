/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import { render } from '@testing-library/react';
import SettingRow from '../../../components/settings/SettingRow';
import { colors } from '../../../components/denali/styles';

describe('SettingRow', () => {
    it('should render switch type setting row', () => {
        let domain = 'domain';
        let role = 'roleName';
        const roleDetails = {
            reviewEnabled: true,
            selfServe: false,
            memberExpiryDays: 3,
            serviceExpiryDays: 3,
            tokenExpiryMins: 15,
            certExpiryMins: 15,
        };
        const { getByTestId } = render(
            <table>
                <tbody>
                    <SettingRow
                        roleDetails={roleDetails}
                        domain={domain}
                        role={role}
                        name='test'
                        label='Test'
                        type='switch'
                        desc='desc for testing'
                        value={true}
                        justificationRequired={true}
                    />
                </tbody>
            </table>
        );

        const settingRow = getByTestId('setting-row');

        expect(settingRow).toMatchSnapshot();
    });
    it('should render input type setting row', () => {
        let domain = 'domain';
        let role = 'roleName';
        const roleDetails = {
            reviewEnabled: true,
            selfServe: false,
            memberExpiryDays: 3,
            serviceExpiryDays: 3,
            tokenExpiryMins: 15,
            certExpiryMins: 15,
        };
        const { getByTestId } = render(
            <table>
                <tbody>
                    <SettingRow
                        roleDetails={roleDetails}
                        domain={domain}
                        role={role}
                        name='test'
                        label='Test'
                        type='input'
                        desc='desc for testing'
                        value={'15'}
                        justificationRequired={true}
                    />
                </tbody>
            </table>
        );

        const settingRow = getByTestId('setting-row');

        expect(settingRow).toMatchSnapshot();
    });
});
