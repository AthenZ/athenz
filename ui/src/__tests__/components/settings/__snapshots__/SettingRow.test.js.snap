// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingRow should render input type setting row 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin-top: 10px;
  box-shadow: 0 1px 4px #d9d9d9;
  border: 1px solid #fff;
  -webkit-border-image: none;
  border-image: none;
  -webkit-border-image: initial;
  border-image: initial;
  height: 50px;
}

.emotion-2 {
  text-align: left;
  width: auto;
  padding: 5px 0 5px 15px;
  vertical-align: middle;
  word-break: break-all;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 300;
  position: relative;
  width: 100%;
}

.emotion-8 input {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: rgba(53,112,244,0.05);
  border-top: 2px solid transparent;
  border-bottom: 2px solid transparent;
  border-right: transparent;
  border-left: transparent;
  border-radius: 2px;
  box-shadow: none;
  box-sizing: border-box;
  color: #303030;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  font: inherit;
  height: 36px;
  margin: 0;
  outline: none;
  padding: 0 1rem;
  text-align: left;
  width: 100%;
}

.emotion-8.animated input {
  -webkit-transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out,border 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out,border 0.2s ease-in-out;
}

.emotion-8 input::-webkit-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-8 input::-moz-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-8 input:-ms-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-8 input::placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-8 input.focused,
.emotion-8 input:active,
.emotion-8 input:focus {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #3570f4;
  color: #303030;
}

.emotion-8 input:invalid {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #d01111;
  color: #303030;
}

.emotion-8 .message {
  font-size: 12px;
  left: 0;
  line-height: 1.8;
  position: absolute;
  top: 2.571rem;
}

.emotion-8.error input,
.emotion-8.error input.focused,
.emotion-8.error input:active,
.emotion-8.error input:focus {
  border-bottom: 2px solid #d01111;
}

.emotion-8.error .message {
  color: #d01111;
}

.emotion-9 {
  margin-top: 5px;
}

<tr
  class="emotion-0 emotion-1"
  data-testid="setting-row"
>
  <td
    class="emotion-2 emotion-3"
    width="auto"
  >
    Test
  </td>
  <td
    class="emotion-2 emotion-3"
    width="auto"
  >
    <div
      class="emotion-6 emotion-7"
    >
      <div
        class="emotion-8 denali-input animated emotion-9 emotion-10"
        data-testid="input-wrapper"
      >
        <input
          autocomplete="off"
          data-testid="input-node"
          id="setting-test"
          pattern="[0-9]*"
          value="15"
        />
        <div
          class="message"
          data-testid="message"
        />
      </div>
    </div>
  </td>
  <td
    class="emotion-2 emotion-3"
    width="auto"
  >
    desc for testing
  </td>
</tr>
`;

exports[`SettingRow should render switch type setting row 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin-top: 10px;
  box-shadow: 0 1px 4px #d9d9d9;
  border: 1px solid #fff;
  -webkit-border-image: none;
  border-image: none;
  -webkit-border-image: initial;
  border-image: initial;
  height: 50px;
}

.emotion-2 {
  text-align: left;
  width: auto;
  padding: 5px 0 5px 15px;
  vertical-align: middle;
  word-break: break-all;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 18px;
  outline: 0;
  position: relative;
}

.emotion-6>input[type='checkbox'] {
  box-sizing: border-box;
  cursor: pointer;
  position: absolute;
  top: 0;
  visibility: hidden;
}

.emotion-6>input[type='checkbox']+label {
  box-sizing: border-box;
  cursor: pointer;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 300;
  height: 18px;
  line-height: 18px;
  white-space: nowrap;
}

.emotion-6>input[type='checkbox']+label .label-content {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: 6px;
}

@-moz-document url-prefix() {
  .emotion-6>input[type='checkbox']+label .label-content {
    margin-top: 1px;
  }
}

.emotion-6>input[type='checkbox']+label:before {
  background: #d5d5d5;
  border-radius: 9px;
  box-sizing: border-box;
  content: '';
  cursor: pointer;
  display: inline-block;
  height: 4px;
  line-height: 16px;
  margin: 7px 0 0 0;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  vertical-align: top;
  width: 34px;
}

.emotion-6>input[type='checkbox']+label::after {
  background-color: #fff;
  border: 2px solid rgba(54,151,242,0.5);
  border-radius: 50%;
  box-sizing: border-box;
  content: ' ';
  cursor: pointer;
  display: block;
  height: 18px;
  left: 0;
  position: absolute;
  -webkit-transition: all 0.25s ease;
  transition: all 0.25s ease;
  top: 0;
  width: 18px;
}

.emotion-6>input[type='checkbox']:checked+label::before {
  background-color: rgba(21,192,70,0.45);
}

.emotion-6>input[type='checkbox']:checked+label::after {
  background-color: #15c046;
  border-color: #15c046;
  left: 16px;
}

.emotion-6>input[type='checkbox']:disabled+label {
  color: #9a9a9a;
  cursor: not-allowed;
}

.emotion-6>input[type='checkbox']:disabled+label::before {
  background-color: #e8e8e8;
  cursor: not-allowed;
}

.emotion-6>input[type='checkbox']:disabled+label::after {
  border-color: #d5d5d5;
  cursor: not-allowed;
}

.emotion-6>input[type='checkbox']:checked:disabled+label::after {
  background: #d5d5d5;
}

<tr
  class="emotion-0 emotion-1"
  data-testid="setting-row"
>
  <td
    class="emotion-2 emotion-3"
    width="auto"
  >
    Test
  </td>
  <td
    class="emotion-2 emotion-3"
    width="auto"
  >
    <div
      class="emotion-6 denali-switch"
      data-testid="switch-wrapper"
    >
      <input
        checked=""
        data-testid="settingtest-switch-input"
        id="switch-settingtest"
        name="settingtest"
        type="checkbox"
        value="true"
      />
      <label
        for="switch-settingtest"
      />
    </div>
  </td>
  <td
    class="emotion-2 emotion-3"
    width="auto"
  >
    desc for testing
  </td>
</tr>
`;
